server.go

```diff
@@ -52,6 +64,83 @@ func handler(w http.ResponseWriter, r *http.Request) {
 }
 
 func recipesHandler(w http.ResponseWriter, r *http.Request) {
+	// read in form
+	days, _ := strconv.Atoi(r.FormValue("days"))
+	breakfast := r.FormValue("breakfast") == "on"
+	lunch := r.FormValue("lunch") == "on"
+	dinner := r.FormValue("dinner") == "on"
+	dessert := r.FormValue("dessert") == "on"
+
+	// sum number of recipes
+	numRecipes := 0
+	if breakfast {
+		numRecipes += days
+	}
+	if lunch {
+		numRecipes += days
+	}
+	if dinner {
+		numRecipes += days
+	}
+	if dessert {
+		numRecipes += days
+	}
+	// limit the size to that of the DB
+	if len(database) < numRecipes {
+		numRecipes = len(database)
+	}
+	recipes = make([]Recipe, 0, 0)
+
+	// filter out repeats
+	var filters []func(Recipe) bool
+	filters = append(filters, func(rec Recipe) bool {
+		return !containsRecipe(recipes, rec)
+	})
+
+	// breakfast
+	for i := 0; (i < days) && breakfast; i++ {
+		// get random recipe
+		mealfilter := append(filters, func(rec Recipe) bool {
+			return contains(rec.Attributes, "breakfast")
+		})
+		recipe := getRecipe(mealfilter)
+		recipes = append(recipes, recipe)
+	}
+	// lunch
+	for i := 0; (i < days) && lunch; i++ {
+		// get random recipe
+		mealfilter := append(filters, func(rec Recipe) bool {
+			return contains(rec.Attributes, "lunch")
+		})
+		recipe := getRecipe(mealfilter)
+		recipes = append(recipes, recipe)
+	}
+	// dinner
+	for i := 0; (i < days) && dinner; i++ {
+		// get random recipe
+		mealfilter := append(filters, func(rec Recipe) bool {
+			return contains(rec.Attributes, "dinner")
+		})
+		recipe := getRecipe(mealfilter)
+		recipes = append(recipes, recipe)
+	}
+	// dessert
+	for i := 0; (i < days) && dessert; i++ {
+		// get random recipe
+		mealfilter := append(filters, func(rec Recipe) bool {
+			return contains(rec.Attributes, "dessert")
+		})
+		recipe := getRecipe(mealfilter)
+		recipes = append(recipes, recipe)
+	}
+
+	/*
```

Comment:
commented out code more like commented out NO'd

Reply:
DAAAAAAAAAAAAAAAAAAAAMN. You're so right, my dude.